src/analysis/excluded_terms_in_mondo/excluded_terms_in_mondo.py

```diff
@@ -0,0 +1,196 @@
+
+Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/31
+- GitHub PR: https://github.com/monarch-initiative/mondo-ingest/pull/35
+"""
+import os
+import sys
+from argparse import ArgumentParser
+from typing import Dict, List
+
+import pandas as pd
+
+
+# Vars
+# # Static
+THIS_SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+ANALYSIS_DIR = os.path.join(THIS_SCRIPT_DIR, '..')
+PROJECT_DIR = os.path.join(ANALYSIS_DIR, '..', '..')
+MAPPINGS_DIR = os.path.join(THIS_SCRIPT_DIR, 'ignored', 'mappings')
+REPORTS_DIR = os.path.join(PROJECT_DIR, 'src', 'ontology', 'reports')
+SIGNATURE_FILES_DIR = REPORTS_DIR
+OUTDIR = REPORTS_DIR
+# # Config
+CONFIG = {
+    # todo: get list from mondo config yml:
+    'ontologies': ['doid', 'icd10cm', 'icd10who', 'ncit', 'omim', 'ordo'],
+    'mondo_mappings_path': os.path.join(MAPPINGS_DIR, 'mondo.sssom.tsv'),
+    'component_sig_pattern': 'component_signature-{}.tsv',
+    'mirror_sig_pattern': 'mirror_signature-{}.tsv',
+    # outpath: summary will be saved in same dir, with _summary at end of name
+    'outpath': os.path.join(OUTDIR, 'excluded_terms_xrefed_in_mondo.tsv'),
+    'save': True,  # save dataframes to disk
+}
+
+
+# Functions
+def uri_to_curie(uri: str) -> str:
+    """Takes an ontological URI and returns a CURIE. Works on the following patterns:
+    - http://.../PREFIX/CODE
+    - http://.../PREFIX_CODE"""
+    uri_list: List[str] = uri.split('/')
+    end_bit = uri_list[-1]
+    end_bit = end_bit.replace('>', '')  # sometimes URIs are wrapped in angle brackets <>
+    if '_' in end_bit:
+        curie = end_bit.replace('_', ':')
+    else:
+        curie = uri_list[-2] + ':' + end_bit
+    return curie
+
+
+def load_and_format_tsv(onto_name: str, pattern: str, _dir: str = SIGNATURE_FILES_DIR) -> pd.DataFrame:
+    """Load and format"""
+    # Get path
+    path = os.path.join(_dir, pattern.format(onto_name))
+    # Read
+    df = pd.read_csv(path, sep='\t').fillna('')
+    # Format
+    df['?term'] = df['?term'].apply(lambda x: x.strip())
+    df['?term'] = df['?term'].apply(uri_to_curie)
+    df = df.rename({'?term': 'term_id'})
+    return df
+
+
+def run(
+    ontologies: List[str], mondo_mappings_path: str, component_sig_pattern: str, mirror_sig_pattern: str,
+    outpath: str, save=CONFIG['save']
+) -> Dict[str, pd.DataFrame]:
+    """Wrapper to run for all ontologies"""
+    onto_field = 'ontology'
+    mirror_field = '1_in_mirror_tsv'
+    component_field = '2_in_component_tsv'
+    mappings_field = '3_in_mondo_xrefs'
+    analytical_field = 'in1_notIn2_in3'
+    summary_count_field = 'n_in1_notIn2_in3'
+    summary_outpath = outpath.replace('.tsv', '_summary.tsv')
+    mondo_mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#').fillna('')
+    mondo_mappings_set = set(mondo_mappings_df['object_id'])
+
+    # todo?: Do we want only (a) certain prefixes? Or (b) just any prefix in the corresponding mirror/component file?
```

Comment:
@matentzn @nicolevasilevsky @sabrinatoro `# todo?: Do we want only (a) certain prefixes? Or (b) just any prefix in the corresponding mirror/component file? If (b): Make map onto_names::prefix: DOID, (OMIM, OMIMPS), Orphanet, NCIT, ICD10CM ICD10WHO, SNOMED`

What I mean by this is, let's say that I'm analyzing NCIT but it happens to have some PATO terms that appear in it (just for sake of argument). Should I ignore those terms and also any other terms that have a prefix other than `NCIT:`?

Right now I'm including _all_ terms in the analysis, regardless of prefix.

Now that I think about it, I probably need to change this...

Is there a list of prefixes by ontology that I can access? I now that OMIM has `OMIM:` and `OMIMPS:`, but I'm not sure if the other ontologies have more than 1 prefix associated with their own terms.

Reply:
Very good question! I think we should use the central prefix map for this: https://github.com/monarch-initiative/mondo-ingest/blob/main/src/ontology/metadata/mondo.sssom.config.yml

Reply:
Hmm I didn't think of using the prefix map for this. But I see that I technically can.

Basically I was looking for something like this (a):
```yml
  OMIM:
    - https://omim.org/entry/
    - http://identifiers.org/omim/
    - http://purl.obolibrary.org/obo/OMIM_
    - http://omim.org/entry/
```

Rather than this, which is what is currently in the yml (b):
```yml
  OMIM: https://omim.org/entry/
  OMIM__2: http://identifiers.org/omim/
  OMIM__3: http://purl.obolibrary.org/obo/OMIM_
  OMIM__4: http://omim.org/entry/
```

However, I can compute (b) from (a), so I'll try that.

Reply:
This has been done using the individual `.yml` config file for each respective ontology.